File: tests/matcher_test.py
from pathlib import Path
from src.matcher import Matcher

def test_missing() -> None:
    matcher = Matcher(Path("src/valid-words.txt"))
    assert matcher.matches("^zzzzz$") == ()

def test_exact() -> None:
    matcher = Matcher(Path("src/valid-words.txt"))
    assert matcher.matches("^seria$") == ("seria",)

def test_multiple() -> None:
    matcher = Matcher(Path("src/valid-words.txt"))
    assert matcher.matches("^able[dr]$") == ("abled", "abler")

File: src/test/kotlin/waffle/engine/MatcherTest.kt
package waffle.engine

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class MatcherTest {
    @Test
    fun `no matches`() {
        val matches = Matcher.search("^zzzzz$")
        assertThat(matches).isEmpty()
    }

    @Test
    fun `single match`() {
        val matches = Matcher.search("^seria$")
        assertThat(matches).containsExactlyInAnyOrder("seria")
    }

    @Test
    fun `multiple match`() {
        val matches = Matcher.search("^able[dr]$")
        assertThat(matches).containsExactlyInAnyOrder("abled", "abler")
    }
}